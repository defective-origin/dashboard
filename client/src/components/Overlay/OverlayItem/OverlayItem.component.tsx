import React, { forwardRef } from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---

// ---| self |---
import css from './OverlayItem.module.scss'

export type OverlaySideVariant = 'left' | 'right' | 'top' | 'bottom'
export type OverlaySideCernerVariant = 'left-center' | 'right-center' | 'top-center' | 'bottom-center'
export type OverlayCernerVariant = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'
export type OverlayCenterVariant = 'center' | 'full'
export type OverlayVariant = OverlaySideVariant | OverlaySideCernerVariant | OverlayCernerVariant | OverlayCenterVariant

export type OverlayItemProps = React.HTMLAttributes<HTMLDivElement> & {
  v?: OverlayVariant
  width?: number | string
  height?: number | string
  indent?: number | string
}

/**
 * Overlay placed item.
 *
 * How to use
 * @example
 * <OverlayItem v="center" indent={10} />
 */
export const OverlayItem = forwardRef<HTMLDivElement, OverlayItemProps>((props, ref): JSX.Element => {
  const {
    v = 'center',
    width,
    height,
    indent,
    style,
    className,
    ...otherProps
  } = props
  const _className = cn(css.OverlayItem, css[v], className)
  const styles = {
    ...style,
    margin: indent,
    width: v === 'full' ? undefined : width,
    height: v === 'full' ? undefined : height,
  }

  return <div ref={ref} className={_className} style={styles} {...otherProps} />
})

OverlayItem.displayName = 'OverlayItem'

export default OverlayItem
