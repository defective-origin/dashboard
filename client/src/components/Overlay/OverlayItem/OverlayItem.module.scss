@import 'src/theme';

:root {
  // --ui-framework-color: var(--account-framework-color);
}

.OverlayItem {
  display: grid;
  position: absolute;
  min-width: 0;
  min-height: 0;
  pointer-events: all;
}

// side variants
.left { top: 0; bottom: 0; left: 0; }
.right { top: 0; bottom: 0; right: 0; }
.top { right: 0; left: 0; top: 0; }
.bottom { right: 0; left: 0; bottom: 0; }

// side-center variants
.left-center { top: 0; bottom: 0; left: 0; }
.right-center { top: 0; bottom: 0; right: 0; }
.top-center { right: 0; left: 0; top: 0; }
.bottom-center { right: 0; left: 0; bottom: 0; }

// cerner variants
.top-left { top: 0; left: 0; }
.top-right { top: 0; right: 0; }
.bottom-left { bottom: 0; left: 0; }
.bottom-right { bottom: 0; right: 0; }

// center variants
.full { inset: 0; }
.center { inset: 0; margin: auto; }
