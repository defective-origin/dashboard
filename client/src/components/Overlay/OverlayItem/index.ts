/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './OverlayItem.component'

/* INJECT_EXPORT_PLACE */
export * from './OverlayItem.component'
