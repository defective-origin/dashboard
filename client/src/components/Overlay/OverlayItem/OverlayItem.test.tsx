// ---| tests |---
import { render } from 'tests'

// ---| self |---
import OverlayItem from './OverlayItem.component'
// import useOverlayItem from './OverlayItem.hook'
// import OverlayItemProvider, OverlayItem from './OverlayItem.context'

describe('[OverlayItem] component', () => {
  it('should render component', () => {
    const container = render(<OverlayItem />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})

// describe('[OverlayItem] hook', () => {
//   it('should return options', () => {
//     const { result } = renderHook(() => useOverlayItem())
//
//     expect(result.current).toEqual(null)
//   })
// })

// describe('[OverlayItem] context', () => {
//   const wrapper = (props: OverlayItemProps) => <OverlayItemProvider { ...props } />
//
//   it('should return options', () => {
//     const { result } = renderHook(() => useOverlayItem(), { wrapper })
//
//     expect(result.current).toBeTruthy()
//   })
// })
