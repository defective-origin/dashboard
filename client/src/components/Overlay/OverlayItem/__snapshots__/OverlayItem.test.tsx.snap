// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[OverlayItem] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="_OverlayItem_2ec79c _center_2ec79c"
  />
</DocumentFragment>
`;
